src/octoprint/timelapse.py

```diff
@@ -752,25 +765,23 @@ def _perform_capture(self, filename, onerror=None):
 
         eventManager().fire(Events.CAPTURE_START, {"file": filename})
         try:
-            self._logger.debug(f"Going to capture {filename} from {self._snapshot_url}")
-            r = requests.get(
-                self._snapshot_url,
-                stream=True,
-                timeout=self._snapshot_timeout,
-                verify=self._snapshot_validate_ssl,
+            self._logger.debug(
+                f"Going to capture {filename} from {self._webcam.config.name} provided by {self._webcam.providerIdentifier}"
             )
-            r.raise_for_status()
+            iter = self._webcam.providerPlugin.take_snapshot(self._webcam)
 
             with open(filename, "wb") as f:
-                for chunk in r.iter_content(chunk_size=1024):
+                for chunk in iter:
```

Comment:
```suggestion
                for chunk in snapshot:
```

`iter` is a reserved word, let's not overwrite that.

Reply:
Done